### PR TITLE
Fix for issue #829

### PR DESCRIPTION
**Pull request by arstrube**
First commit, manipulate some tests to provoke the problem & failure.
Second commit, proposed fix.

### CODE REVIEW

**Comment by arstrube** [comment]
@basvodde Travis CI for the pull request doesn't show the intended failures. These can be seen here:

https://travis-ci.org/arstrube/cpputest/builds/103618693

**Comment by basvodde** [comment]
Ok.

File: src/CppUTestExt/MockSupport.cpp [hunk 1]
```diff
@@ -310,20 +310,15 @@ void MockSupport::countCheck()
     UtestShell::getCurrent()->countCheck();
 }
 
-void MockSupport::checkExpectationsOfLastCall()
+void MockSupport::checkExpectations()
 {
+    for(MockNamedValueListNode *p = data_.begin(); p; p = p->next())
+        if(getMockSupport(p) && getMockSupport(p)->lastActualFunctionCall_)
```
Comment (cuitoldfish) [inline src/CppUTestExt/MockSupport.cpp]: Do we still need to check the condition `getMockSupport(p)->lastActualFunctionCall_`? if no lastActualFunctionCall_ then we should not go on checking expectation??

Comment (arstrube) [inline src/CppUTestExt/MockSupport.cpp]: Not sure. I took it out at some point and things started going wrong ;-).

Comment (cuitoldfish) [inline src/CppUTestExt/MockSupport.cpp]: I'm afraid if lastActualFunctionCall is null then we lost chance to check other scope out of order.
for example:
```.cpp
    mock().strictOrder();
    mock("bla").strictOrder();
	mock().ignoreOtherCalls();

    MockExpectedCallsListForTest expectations;
    expectations.addFunction("foo1", 1)->callWasMade(2);
    expectations.addFunction("foo2", 2)->callWasMade(1);
    MockCallOrderFailure expectedFailure(mockFailureTest(), expectations);

    mock("bla").expectOneCall("foo1");
    mock("foo").expectOneCall("foo1");
    mock("foo").expectOneCall("foo2"); 

    mock("bla").actualCall("foo1");
    mock("foo").actualCall("foo2");
    mock("foo").actualCall("unexpected1");
    mock("foo").actualCall("foo1");
    mock("foo").actualCall("unexpected2");
```
I admit i'm fastidious about the out of order check :)

I have another proposed fix pull request, hopefully you guys can give some look and comments.

**Pull request merged**